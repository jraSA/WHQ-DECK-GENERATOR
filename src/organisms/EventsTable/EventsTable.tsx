import React from 'react';
import { ImageBackground } from 'react-native';
import styled from 'styled-components/native';

import eventDeck, { back } from '../../assets/eventDeck';
import shuffle from '../../helpers/shuffle/Shuffle';
import Deck from '../../molecules/Deck';

const DeckContainer = styled.View`
    flex: 1;
    flex-direction: row;
    align-self: center;
`;
const styles = {
    imgBackground: {
        width: '100%',
        height: '100%',
        flex: 1,
    },
};

const EventsTable = (props: { navigation: any }) => {
    return (
        <ImageBackground style={styles.imgBackground} source={require('../../../assets/table.png')}>
            <DeckContainer>
                <Deck generateDeck={() => shuffle(eventDeck)} back={back} />
            </DeckContainer>
        </ImageBackground>
    );
};

export default EventsTable;
