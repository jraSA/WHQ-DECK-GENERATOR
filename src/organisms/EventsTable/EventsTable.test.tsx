import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import EventsTable from './index';

describe('EventsTable Snapshot test', () => {
    it('renders correctly', () => {
        const mockNavigation = jest.fn();
        const tree = renderer.create(<EventsTable navigation={mockNavigation} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
