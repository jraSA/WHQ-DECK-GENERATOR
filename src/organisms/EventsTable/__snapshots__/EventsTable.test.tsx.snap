// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsTable Snapshot test renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
              "width": 233,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 22,
                "height": 360,
                "width": 233,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "borderRadius": 22,
                  "height": 360,
                  "width": 233,
                },
                null,
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#00609e",
                  "borderRadius": 24,
                  "height": 44,
                  "justifyContent": "center",
                  "margin": 7,
                  "width": 44,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 360,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
              "width": 233,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
